src/BuildScriptGenerator/Node/NodePlatform.cs

```diff
@@ -187,11 +187,13 @@ public PlatformDetectorResult Detect(RepositoryContext context)
                 throw new InvalidUsageException("Multiple monorepo package management tools are found, please choose to use either Lerna or Lage.");
             }
 
+            string yarnVersionSpec = null;
```

Comment:
nit: let's assign the value of `yarnVersionSpec` immediately and use it in the following line:


```cs
string yarnVersionSpec = packageJson?.engines?.yarn;
if (ctx.SourceRepo.FileExists(NodeConstants.YarnLockFileName) || yarnVersionSpec != null)
{
    ...
}
```

Reply:
sure.